### PR TITLE
Fix improper signatures

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #841

### CODE REVIEW

File: functest_requirements.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@ pulp-smash @ git+https://github.com/pulp/pulp-smash.git
 orionutils
 pytest
 dynaconf
+ansible @ git+https://github.com/ansible/ansible.git@devel
```
Comment (gerrod3) [inline functest_requirements.txt:5]: The installation signature verification is currently only available on the devel branch.

File: pulp_ansible/app/tasks/signature.py [hunk 2]
```diff
@@ -71,13 +73,20 @@ async def sign_collection_version(self, collection_version):
...
-                async with aiofiles.open(f"{d}/MANIFEST.json", "w") as t:
-                    await t.write(json.dumps(collection_version.manifest))
-                result = await self.signing_service.asign(t.name)
-                async with aiofiles.open(result["signature"], "rb") as sig:
-                    data = await sig.read()
+            # OpenPGP doesn't take filename into account for signatures, not sure about others
+            async with aiofiles.tempfile.NamedTemporaryFile(dir=".", mode="wb") as m:
+                cartifact = await sync_to_async(
+                    collection_version.contentartifact_set.select_related("artifact").first
+                )()
```
Comment (gerrod3) [inline pulp_ansible/app/tasks/signature.py:80]: Not sure if there is a better way to get this content_artifact to extract the artifact.

Comment (mdellweg) [reply pulp_ansible/app/tasks/signature.py:80]: No, this looks right to me.

File: pulp_ansible/app/tasks/signature.py [hunk 3]
```diff
@@ -71,13 +73,20 @@ async def sign_collection_version(self, collection_version):
...
+                artifact_name = cartifact.artifact.file.name
+                artifact_file = storage.open(artifact_name)
+                with tarfile.open(fileobj=artifact_file, mode="r") as tar:
+                    man = get_file_obj_from_tarball(tar, "MANIFEST.json", artifact_name)
+                    await m.write(man.read())
```
Comment (mdellweg) [inline pulp_ansible/app/tasks/signature.py:85]: Should we do some more sync_to_async around tar file operations?

**Review by mdellweg — COMMENTED** [review]
I am wondering, whether we should verify all file digests in the manifest before actually signing it.

**Review by rochacbruno — APPROVED** [review]

File: functest_requirements.txt (hunk 1 above)
Comment (mdellweg) [reply functest_requirements.txt]: To be upfront: This is only a test requirement. So we are fine with relying on some unreleased piece of code.

**Comment by gerrod3** [comment]
> I am wondering, whether we should verify all file digests in the manifest before actually signing it.

We could, it would slow down the task but maybe it is something to look into for the future. I'm not sure I can look into adding it now.

**Pull request merged**